packages/@uppy/companion/src/server/helpers/utils.js

```diff
@@ -1,5 +1,6 @@
 const crypto = require('node:crypto')
 
+const ivLength = 12
```

Comment:
Since you're using this with HKDF, you can actually use a larger nonce here. (For example, 256 bits or 32 bytes.)

You can then synthetically derive the GCM nonce from the HKDF output. This is what PASETO v3/v4 do, if you need an incumbent design.

https://github.com/paseto-standard/paseto-spec/blob/master/docs/01-Protocol-Versions/Version3.md#encrypt
(See `n2`).

Reply:
Thank you! This makes a lot of sense. I should have thought about that :sweat_smile: I'll amend the PR.